libflux/src/semantic/types.rs

```diff
@@ -119,14 +119,28 @@ impl fmt::Display for MonoType {
 // Tvar stands for type variable.
 // A type variable holds an unknown type.
 #[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
-pub struct Tvar(i64);
+pub struct Tvar(pub i64);
 
 impl fmt::Display for Tvar {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
         write!(f, "t{}", self.0)
     }
 }
 
+// Fresher returns a fresh type variable with incrementing id.
+pub struct Fresher(i64);
+
+impl Fresher {
+    pub fn new() -> Fresher {
+        Fresher(0)
+    }
+
+    pub fn get(&mut self) -> Tvar {
```

Comment:
Could you name this `fresh` instead of `get`? It's just a little more explicit about what is actually happening. I usually don't expect a `get` method to mutate the object.